doc/specification/language-proposal.md

```diff
@@ -0,0 +1,1518 @@
+
+
+### Whitespace
+<a id="whitespace"/>
+
+In some expressions, where it denotes multiplication, whitespace is mandatory, but otherwise whitespace is not significant. It comprises any non-empty sequence of characters with the White_Space property.
+
+> ***whitespace*** ::= {White_Space}\+
+
+#### Token value
+
+A ***whitespace*** token has no value.
+
+
+### Symbols
+<a id="symbols"/>
+
+Symbols correspond to Arblang identifiers and keywords. Arblang keywords are generally contextual: the interpretation of a symbol as an identifier or a keyword is determined in parsing.
+
+> ***symbol*** ::= _symbol-start_ _symbol-continue_\* | _symbol-degree-prefixed_
+>
+> _symbol-start_ ::= {L} | {Nl} | {Other_ID_Start}
+>
+> _symbol-continue_ ::= _symbol-start_ | {Nd} | {Mn} | {Mc} | {Pc} | _prime-mark_
+>
+> _prime-mark_ ::= U+0027 `'` APOSTROPHE | U+02B9 `ʹ` MODIFIER LETTER PRIME | U+2032 `′` PRIME
+>
+> _symbol-degree-prefixed_ ::= U+00B0 `°` DEGREE SIGN {L}+ | U+2109 `℉` DEGREE FAHRENHEIT | U+2109 `℃` DEGREE CENTIGRADE
+
+#### Token value
+
+Canonicalization substitutes any character in _prime-mark_ with U+0027 APOSTROPHE after performing NFKC normalization.
+
+#### Notes
+
+Identifiers broadly follow the conventions of Python, except:
+* `_` is not a valid _symbol-start_.
+* A ***symbol*** may contain _prime-mark_ characters.
+* Special degree-sign prefixed symbols are permitted, in order to represent temperature scales. Compatibility characters U+2109 `℉` and U+2109 `℃` will be converted to `°F` and `°C` respectively in NFKC normalization.
+* Characters that are only _compatibility_ equivalent to an acceptable _symbol_ character are not permitted. In particular, superscript numerals are not permitted within a _symbol_.
+
+
+### Superscript literals
+<a id="superscript-literals"/>
+
+A superscript literal is used to denote exponents in the absence of the exponentiation operator.
+
+> ***superscript-literal*** ::= _superscript-minus_? _superscript-digit_+
+>
+> _superscript-minus_ ::= U+207B `⁻` SUPERSCRIPT MINUS
+>
+> _superscript-digit_ ::= `⁰` | `¹` | `²` | `³` | `⁴` | `⁵` | `⁶` | `⁷` | `⁸` | `⁹`
+
+#### Token value
+
+After NFKC normalization, which takes superscript digits and minus to regular digits and U+2212 MINUS SIGN, the token value is canonicalized by mapping U+2212 to U+002d `-` HYPHEN-MINUS.
+
+
+### Numeric literals
+<a id="numeric-literals"/>
+
+Numeric literals represent both fractional and integer literal values.
+
+> ***numeric-literal*** ::= _digit-sequence_ (_decimal-separator_ _digit-sequence_)? _exponent_?
+>
+> _digit-sequence_ ::= ( `0` | `1` | `2` | `3` | `4` | `5` | `6` | `7` | `8` | `9` )+ ( _digit-group-separator_ _digit_sequence_)?
+>
+> _digit-group-separator_ ::= U+0027 `'` APOSTROPHE | _whitespace_
```

Comment:
Potential issue: `U+0027` is used in identifiers as well.

Reply:
It is unambiguous because it can only occur between digits inside a numeric literal, which itself cannot start with something that be used to start an identifier (or symbol) token.

Reply:
What about `a'100`?

Reply:
Tokenized as a ***symbol***.